src/main/java/seedu/address/model/loan/LoanRate.java

```diff
@@ -0,0 +1,53 @@
+package seedu.address.model.loan;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Loan's rate in the loan book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidRate}
+ */
+public class LoanRate {
+
+    public static final String LOANRATE_VALIDATION_REGEX =
+            "^(([1-9]\\d*|0)|(([0]\\.\\d{1,2}|[1-9][0-9]*\\.\\d{1,2})))$";
+    public static final String MESSAGE_LOANRATE_CONSTRAINTS =
+            "Rate should be not less than 0 and have at most 2 decimal places.";
+
+    public final double rate;
+
+    /**
+     * Constructs a {@code LoanRate}.
+     *
+     * @param rate A valid rate.
+     */
+    LoanRate(String rate) {
+        requireNonNull(rate);
+        checkArgument(isValidRate(rate), MESSAGE_LOANRATE_CONSTRAINTS);
+        this.rate = Double.parseDouble(rate);
+    }
+
+    /**
+     * Returns true if a given string is a valid rate.
+     */
+    public static boolean isValidRate(String test) {
+        return test.matches(LOANRATE_VALIDATION_REGEX);
+    }
+
+    @Override
+    public String toString() {
+        return String.valueOf(rate);
```

Comment:
Might want to consider truncation of decimal digits in this method...

Reply:
You mean 2.0 -> 2 ? Like this?

Reply:
More like 3.3333333 -> 3.33 . It is not important at the moment, we can deal with this when we meet this issue.

Reply:
No, the input is alr restricted to have less than 2 decimal places. So 3.3333 is not a valid input at the very beginning. That's why I did not consider this issue here.

Reply:
Oh, I didn't realise you have changed the regex in one of your recent commits...

My concern is when we are displaying the rates, since some numbers cannot be represented in binary float exactly.